pkg/admission/server_test.go

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/lithammer/dedent"
+	"github.com/stretchr/testify/assert"
+	admission "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var decoder = codecs.UniversalDeserializer()
+
+func TestServeHTTPInvalidBody(t *testing.T) {
+	assert := assert.New(t)
+	res := httptest.NewRecorder()
+	handler := http.HandlerFunc(ServeHTTP)
+	req, err := http.NewRequest("POST", "", nil)
+	req = req.WithContext(context.Background())
+	assert.Nil(err)
+	handler.ServeHTTP(res, req)
+	assert.Equal(400, res.Code)
+	assert.Equal("admission review object is missing\n",
+		res.Body.String())
+}
+
+func TestServeHTTPInvalidMethod(t *testing.T) {
+	assert := assert.New(t)
+	res := httptest.NewRecorder()
+	handler := http.HandlerFunc(ServeHTTP)
+	req, err := http.NewRequest("GET", "", nil)
+	req = req.WithContext(context.Background())
+	assert.Nil(err)
+	handler.ServeHTTP(res, req)
+	assert.Equal(http.StatusMethodNotAllowed, res.Code)
+	assert.Equal("invalid method GET, only POST requests are allowed\n",
+		res.Body.String())
+}
+
+func TestServeHTTPSubmissions(t *testing.T) {
+	for _, apiVersion := range []string{
+		"admission.k8s.io/v1beta1",
```

Comment:
Do we still need to accept this if we use only setup [webhook v1 validation](https://github.com/kubernetes-sigs/gateway-api/pull/617/files#diff-267c12189c93f1a7a1a0163c9f10522d8f2fe0792223c211200f09f941d5c3ddR22-R23)

Reply:
Updated to v1 only.